### DISCUSSION TITLE
Should the logic for when secondary location cards be tweaked?

### DISCUSSION THREAD

**Discussion by elydpg** [breezy-weather/breezy-weather#1101] [General]
I'm specifically asking about this in general discussions thread as it specifically says that changes to the update process fall under the category of "Features that will not be implemented". It currently says in the [weather update process doc](https://github.com/breezy-weather/breezy-weather/blob/main/docs/UPDATES.md) that a manual weather update happens when:

> From your location list, you access a location that has not been refreshed for more than your background update refresh rate (or if set to never, 1.5 hours)

To me, there seems to be an inconsistency here: If I set my background update refresh rate to say 6 hours, then accessing a location whose last update was 5 hours ago would not trigger an update. However, if I set it to never then it would trigger an update, even though background updates are less frequent (I presume "never" is less frequent than "every 6 hours"). Seems a bit odd to me that making background updates less frequent would make manual updates more frequent. To rectify this, I thought of a few alternatives:

* If background update is set to anything less frequent than 1.5 hours, then  
* If background update is set to "never" then the manual update happens as if the background update frequency was set to its most infrequent value where updates still occur; i.e. every 24 hours
* A manual update occurs upon accessing a location that has not been refreshed for more than a fixed number of hours, independent of the background update setting.
    * Either split this off into a separate setting or set it to a fixed default

Curious what others think on the matter

**Comment by papjul** [comment]
Thank you for reading carefully everything I documented :)

Setting refresh rate to "Never" is highly discouraged. Please read fully the warning dialog showing when you set it to Never, mentioning the reasons while you probably don't need to set it to Never. Make sure you're up-to-date as it was added recently.

> If background update is set to "never" then the manual update happens as if the background update frequency was set to its most infrequent value where updates still occur; i.e. every 24 hours

People have different needs: see #632 when it was implemented. I implemented it that way because 1) in the first place, the "Never" should not exist ([some weather apps even refuse to implement it](https://github.com/davidtakac/bura/blob/1.5.1/README.md#will-not-be-implemented)) ) 2) to compensate it, I ensure you're presented with up-to-date data when you try to access them. 1.5 hours is the reasonable default value. It can be changed to a different value, if you follow the recommendation of setting a background refresh rate ;)

> A manual update occurs upon accessing a location that has not been refreshed for more than a fixed number of hours, independent of the background update setting.

> Either split this off into a separate setting or set it to a fixed default

~~That's how it is now: it's a fixed value of 1.5 hours.~~ Nevermind, you said "independent of the background update setting" but my point is still valid: Originally (and it is still used for that purpose), the automatic update on foreground is to compensate background tasks being killed by some manufacturers. So it must follow background update setting. And there won't be an added preference for the Never-use case which is discouraged.

**Comment by elydpg** [comment]
> Setting refresh rate to "Never" is highly discouraged. Please read fully the warning dialog showing when you set it to Never, mentioning the reasons while you probably don't need to set it to Never. Make sure you're up-to-date as it was added recently.

Thanks for letting me know about this! I never actually read it till now because my use case requires background updates more frequent than the default (every hour or half hour). It seems that the main side effect of disabling background updates is that it disables alert notifications. I assume that's because these are fetch and not push notifications, which is a very sensible choice for a small foss weather app. 

> People have different needs: see #632 when it was implemented. I implemented it that way because 1) in the first place, the "Never" should not exist ([some weather apps even refuse to implement it](https://github.com/davidtakac/bura/blob/1.5.1/README.md#will-not-be-implemented)) ) 2) to compensate it, I ensure you're presented with up-to-date data when you try to access them. 1.5 hours is the reasonable default value. It can be changed to a different value, if you follow the recommendation of setting a background refresh rate ;)

From reading #632 it seems that this feature was added for people who want up-to-date data when they open the app but don't want it running in the background at all? Personally this is the behaviour I'd want if I were using the app like a static weather website (which always serves the most up to date data but only when you go to it), so maybe the most suitable choice for users who disable background updates is to run the foreground updates on the most frequent update setting of every 30 minutes?

Of course, if the "never" setting didn't exist, most flavours of Android let you block specific from using mobile data (if not also wifi) in the background, so that could have been a workaround for people who want that static functionality. I'm not sure what the app does if blocked access to internet at the system level though.

**Reply by papjul** [reply]
> maybe the most suitable choice for users who disable background updates is to run the foreground updates on the most frequent update setting of every 30 minutes?

They only want fetches when they use the app. Every 30 min is the opposite of what they want. But I must say they are just lazy to pull to refresh.

> I'm not sure what the app does if blocked access to internet at the system level though.

The background work is delayed until network access is restored (as soon as it is restored, if it's been more than the refresh rate setting).

> my use case requires background updates more frequent than the default (every hour or half hour)

I'm confused now, why are we discussing the Never setting then? 😅

**Reply by elydpg** [reply]
> They only want fetches when they use the app. Every 30 min is the opposite of what they want. But I must say they are just lazy to pull to refresh.

I mean "every 30 minutes" would apply only to the foreground refresh if the background refresh is disabled, right? I guess really the most logical thing is of background refresh is off just don't refresh at all except for on pull to refresh (or on first adding the location, if the sources change, etc.) but you're right that they just don't wanna pull to refresh 😛

> I'm confused now, why are we discussing the Never setting then? 😅

Why not? I like considering the different use cases 🤷
